addon/v1alpha1/types_managedclusteraddon.go

```diff
@@ -29,8 +29,37 @@ type ManagedClusterAddOn struct {
 	Status ManagedClusterAddOnStatus `json:"status"`
 }
 
-// ManagedClusterAddOnSpec is empty for now.
+// ManagedClusterAddOnSpec defines the install configuration of
+// addon agent on managed cluster.
 type ManagedClusterAddOnSpec struct {
```

Comment:
would you add the `leaseDurationSeconds` on the spec to control the addon lease update frequency, by default, it is 60s, and would you add addon available condition type (`ManagedClusterAddOnConditionAvailable`)